src/connection.js

```diff
@@ -331,6 +361,12 @@ class Connection extends EventEmitter {
           return this.request.error.lineNumber = token.lineNumber;
         }
       } else {
+        const isLoginErrorTransient = this.transientErrorLookup.isTransientError(token.number);
+        if (isLoginErrorTransient && this.curTransientRetryCount !== this.config.options.maxRetriesOnTransientErrors) {
+          this.debug.log('Initiating retry on transient error = ', token.number);
+          return this.transitionTo(this.STATE.TRANSIENT_FAILURE_RETRY);
```

Comment:
The "return" statements with a value are an artefact of the CoffeeScript to JavaScript conversion and should not be used whereit makes no sense.